solidity/contracts/bridge/Bridge.sol

```diff
@@ -74,15 +92,37 @@ contract Bridge {
     struct DepositInfo {
         // Ethereum depositor address.
         address depositor;
-        // Deposit amount in satoshi (8-byte LE). For example:
-        // 0.0001 BTC = 10000 satoshi = 0x1027000000000000
-        bytes8 amount;
+        // Deposit amount in satoshi.
+        uint64 amount;
         // UNIX timestamp the deposit was revealed at.
         uint32 revealedAt;
         // Address of the tBTC vault.
         address vault;
+        // UNIX timestamp the deposit was swept at. Note this is not the
+        // time when the deposit was swept on Bitcoin chain but actually
+        // the time when the sweep proof was delivered on Ethereum chain.
+        uint32 sweptAt;
+    }
+
+    /// @notice Represents an info about a sweep.
+    struct SweepInfo {
+        // Hash of the sweep transaction.
+        bytes32 txHash;
+        // Amount of the single transaction output.
+        uint256 amount;
     }
 
+    /// @notice Confirmations on the Bitcoin chain required to successfully
+    ///         evaluate an SPV proof.
+    uint256 public immutable txProofDifficultyFactor;
+
+    /// @notice Address of the Bank this Bridge belongs to.
+    Bank public immutable bank;
```

Comment:
Bank will probably be updatable. I would left a TODO that we should make it updatable for know.

Reply:
I'm not sure since if I'm not wrong the Bank should be the non-upgradeable element of the system. Can you confirm @pdyraga ?

Reply:
Can we see what is the gas difference if we make it non-immutable and if it is acceptable, start with a non-immutable version and leave a TODO to revisit it later? I am still playing with upgradeability and I see the pros and cons of both options.

Reply:
Making it non-immutable costs `2k`. I'd just leave a TODO for now: https://github.com/keep-network/tbtc-v2/pull/106/commits/4c19bc2367eae0b24a509223dde6e981f70a1bae